src/runtime/task/mod.rs

```diff
@@ -58,11 +58,14 @@ pub(crate) struct Task {
     name: Option<String>,
 
     local_storage: StorageMap,
+
+    // Arbitrarily settable tag which is inherited from the parent.
+    tag: u64,
```

Comment:
Make a `struct Tag(u64)`?

Reply:
I think the intended use is for people to do arithmetic with it. I think part of the point is that we cannot change the representation -- we are giving people 64 bits to encode whatever they want and are committing to that.

Reply:
In my mind, the common use case for tags is to represent something discrete (like, we initially called them "colors") that you only want to compare. I understand that it makes sense to commit to a representation and give more flexibility. With the struct clients can still grab the u64 and do with it whatever they want, they just have to be explicit about it. So I don't see any harm in having some extra protection in the common case.

Reply:
I'm inclined to agree with Bernhard.  Defining an opaque `Tag` type that has from/into conversions to `u64` might be better, as it gives us a bit more freedom to change the internal representation later.